tests/controller_tests/test_retrieve.py

```diff
@@ -0,0 +1,17 @@
+from core.const import EXCEL_MIMETYPE
+
+
+def test_invalid_programme_name(seeded_test_client):
+    response = seeded_test_client.get("/retrieve?reporting_round=3&programme_name=invalid")
+    assert response.status_code == 400
+
+
+def test_invalid_reporting_round(seeded_test_client):
+    response = seeded_test_client.get("/retrieve?reporting_round=1&programme_name=Leaky Cauldron regeneration")
+    assert response.status_code == 400
+
+
+def test_retrieval(seeded_test_client):
+    response = seeded_test_client.get("/retrieve?reporting_round=3&programme_name=Leaky Cauldron regeneration")
+    assert response.status_code == 200
+    assert response.content_type == EXCEL_MIMETYPE
```

Comment:
Is it possible to do some more granular assertions than just the status_code and content_type?